src/cocotb/handle.py

```diff
@@ -909,44 +909,50 @@ def _set_value(
                     )
                     return
 
+                # LogicArray used for checking
                 if value < 0:
-                    value_ = LogicArray.from_signed(
-                        value,
-                        Range(len(self) - 1, "downto", 0),
+                    value_ = str(
+                        LogicArray.from_signed(
+                            value,
+                            Range(len(self) - 1, "downto", 0),
+                        )
                     )
                 else:
-                    value_ = LogicArray.from_unsigned(
-                        value,
-                        Range(len(self) - 1, "downto", 0),
+                    value_ = str(
+                        LogicArray.from_unsigned(
+                            value,
+                            Range(len(self) - 1, "downto", 0),
+                        )
                     )
             else:
                 raise OverflowError(
                     f"Int value ({value!r}) out of range for assignment of {len(self)!r}-bit signal ({self._name!r})"
                 )
 
         elif isinstance(value, str):
-            value_ = LogicArray(value, self.range)
+            # LogicArray used for checking
+            value_ = str(LogicArray(value, self.range))
```

Comment:
Interesting we aren't testing this. I'll add a small test to test_handles for this.